File: examples/python/evm/account.fund.py
# Usage: uv run python evm/account.fund.py

import asyncio
from cdp import CdpClient
from dotenv import load_dotenv

load_dotenv()


async def main():
    async with CdpClient() as cdp:
        account = await cdp.evm.get_or_create_account(name="account")

        fund_operation = await account.fund(
            network="base",
            token="eth",
            amount=500000000000000, # 0.0005 eth
        )

        completed_transfer = await account.wait_for_fund_operation_receipt(
            transfer_id=fund_operation.id,
        )

        print(completed_transfer)


asyncio.run(main()) 

File: examples/typescript/evm/account.fund.ts
// Usage: pnpm tsx evm/account.fund.ts

import { CdpClient } from "@coinbase/cdp-sdk";
import "dotenv/config";
async function main() {
  const cdp = new CdpClient();
  const account = await cdp.evm.getOrCreateAccount({ name: "account" });

  const fundOperation = await account.fund({
    network: "base",
    token: "eth",
    amount: 500000000000000n, // 0.0005 eth
  });


  const completedTransfer = await account.waitForFundOperationReceipt({
    transferId: fundOperation.id,
  });

  console.log(completedTransfer);
}

main().catch(console.error);
